javav2/example_code/s3/src/main/java/com/example/s3/express/S3DirectoriesActions.java

```diff
@@ -0,0 +1,551 @@
+                if (exception != null) {
+                    throw new CompletionException("Error creating bucket: " + bucketName, exception);
+                }
+                logger.info(bucketName + " is ready");
+            });
+    }
+
+    /**
+     * Uploads an object to an Amazon S3 bucket asynchronously.
+     *
+     * @param s3Client     the S3 async client to use for the upload
+     * @param bucketName   the name of the S3 bucket to upload the object to
+     * @param bucketObject the name of the object to be uploaded
+     * @param text         the content to be uploaded as the object
+     */
+    public CompletableFuture<PutObjectResponse> putObjectAsync(S3AsyncClient s3Client, String bucketName, String bucketObject, String text) {
+        PutObjectRequest objectRequest = PutObjectRequest.builder()
+            .bucket(bucketName)
+            .key(bucketObject)
+            .build();
+
+        return s3Client.putObject(objectRequest, AsyncRequestBody.fromString(text))
+            .whenComplete((response, exception) -> {
+                if (exception != null) {
+                    throw new CompletionException("Failed to upload file", exception);
+                }
+            });
+    }
+
+    /**
+     * Creates an AWS IAM access key asynchronously for the specified user name.
+     *
+     * @param userName the name of the IAM user for whom to create the access key
+     * @return a {@link CompletableFuture} that completes with the {@link CreateAccessKeyResponse} containing the created access key
+     */
+    public CompletableFuture<CreateAccessKeyResponse> createAccessKeyAsync(String userName) {
+        CreateAccessKeyRequest request = CreateAccessKeyRequest.builder()
+            .userName(userName)
+            .build();
+
+        return getIAMAsyncClient().createAccessKey(request)
+            .whenComplete((response, exception) -> {
+                if (response != null) {
+                    logger.info("Access Key Created.");
+                } else {
+                    if (exception == null) {
+                        throw new CompletionException("An unknown error occurred while creating access key.", null);
+                    }
+
+                    Throwable cause = exception.getCause();
+                    if (cause instanceof IamException) {
+                        throw new CompletionException("IAM error while creating access key: " + cause.getMessage(), cause);
+                    }
+
+                    throw new CompletionException("Failed to create access key: " + exception.getMessage(), exception);
+                }
+            });
+    }
+
+    /**
+     * Selects an availability zone ID based on the specified AWS region.
+     *
+     * @return A map containing the selected availability zone details, including the zone name, zone ID, region name, and state.
+     */
+    public CompletableFuture<String> selectAvailabilityZoneIdAsync() {
+        // Request available zones
```

Comment:
```suggestion
        // Request available zones.
```

(Style is to use periods at the end.)

Reply:
Fixed